docs/user-guide/kubeadm-reference.md

```diff
@@ -0,0 +1,95 @@
+`--api-advertise-addresses` and `--api-external-dns-names`.
+
+- `--cloud-provider`
+
+Currently, `kubeadm` does not provide autodetection of cloud provider.
+This means that load balancing and persistent volumes are not supported out
+of the box. You can specify a cloud provider using `--cloud-provider`.
+Valid values are the ones supported by `controller-manager`, namely `"aws"`,
+`"azure"`, `"cloudstack"`, `"gce"`, `"mesos"`, `"openstack"`, `"ovirt"`,
+`"rackspace"`, `"vsphere"`.
+
+- `--external-etcd-cafile` etcd certificate authority certificate file
+- `--external-etcd-certfile` etcd client certificate file
+- `--external-etcd-endpoints` comma separated list of etcd endpoints to use
+for an external cluster
+- `--external-etcd-keyfile` etcd client key file
+
+By default, `kubeadm` will deploy a single node etcd cluster on the master
+to store Kubernetes state. This means that any failure on the master node
+will require you to rebuild your cluster from scratch. Currently `kubeadm`
+does not support automatic deployment of a highly available etcd cluster.
+If you would like to use your own etcd cluster, you can override this
+behaviour with `--external-etcd-endpoints`. `kubeadm` supports etcd client
+authentication using the `--external-etcd-cafile`, `--external-etcd-certfile`
+and `--external-etcd-keyfile` flags.
+
+- `--pod-network-cidr`
+
+By default, `kubeadm` does not set node CIDR's for pods and allows you to
+bring your own networking configuration through a CNI compatible network
+controller addon such as [Weave Net](https://github.com/weaveworks/weave-kube),
+[Calico](https://github.com/projectcalico/calico-containers/tree/master/docs/cni/kubernetes/manifests/kubeadm)
+or [Canal](https://github.com/tigera/canal/tree/master/k8s-install/kubeadm).
+If you are using a compatible cloud provider or flannel, you can specify a
+subnet to use for each pod on the cluster with the `--pod-network-cidr` flag.
+This should be a minimum of a /16 so that kubeadm is able to assign /24 subnets
+to each node in the cluster.
+
+- `--service-cidr` (default '100.64.0.0/12')
+
+You can use the `--service-cidr` flag to override the subnet Kubernetes uses to
+assign pods IP addresses. By default, the chosen subnet is reserved for carrier
+grade NAT and is highly unlikely to cause a collision.
+
+- `--service-dns-domain` (default 'cluster.local')
+
+By default, `kubeadm` will deploy a cluster that assigns services with DNS names
+`<service_name>.<namespace>.cluster.local>. You can use the `--service-dns-domain`
+to change the DNS name suffix.
+
+- `--token`
+
+By default, `kubeadm` will automatically generate the token used to initialise
+each new node. If you would like to manually specify this token, you can use the
+`--token` flag. The token must be of the format '<6 character string>.<16 character string>'.
+
+- `--use-kubernetes-version` (default 'v1.4.0') the kubernetes version to initialise
+
+`kubeadm` supports a minimum Kubernetes version of v1.4.0
```

Comment:
Something like this could be better:
```
`kubeadm` was originally built for Kubernetes version **v1.4.0**, older versions are not supported.
With this flag you can try any future version, e.g.  **v1.5.0-beta.1** whenever it comes out (check [releases page](https://github.com/kubernetes/kubernetes/releases)).
```